sparkmagic/sparkmagic/controllerwidget/manageendpointwidget.py

```diff
@@ -30,7 +30,13 @@ def get_existing_endpoint_widgets(self):
 
             # Endpoints
             for url, endpoint in self.endpoints.items():
-                endpoint_widgets.append(self.get_endpoint_widget(url, endpoint))
+                try:
+                    endpoint_widgets.append(self.get_endpoint_widget(url, endpoint))
+                except HttpClientException:
+                    # If we can't reach one of the default endpoints, just skip over it
+                    # silently
+                    if not endpoint.is_default:
```

Comment:
Okay, I don't mind logging it here too, but I just want to make sure it's a good idea. As you guessed, the problem I'm trying to fix is someone copying in a default `example_config.json` and getting strange errors about an endpoint they aren't even aware of.

I think the current behavior is good: if they _did_ explicitly configure an endpoint in `config.json` that cannot be connected to, they silently won't see it in the default endpoint list – then when they explicitly add it, they see the connection error and understand what needs to be fixed. The alternative seems worse.